openquake/hazardlib/calc/disagg.py

```diff
@@ -42,17 +42,37 @@
     'BinData', 'mags, dists, lons, lats, trts, pnes')
 
 
+# yield (rlzi, poe, imt), (iml, pne)
+def _disagg(iml, poes, curve_poes, imls, gsim, rupture, rlzi, imt, imt_str,
+            sctx, rctx, dctx, truncation_level, n_epsilons, disagg_poe):
+    if iml is None:  # compute the IMLs from the given poes
+        for poe in poes:
+            iml = numpy.interp(poe, curve_poes, imls)
+            with disagg_poe:
+                [poes_given_rup_eps] = gsim.disaggregate_poe(
+                    sctx, rctx, dctx, imt, iml, truncation_level, n_epsilons)
+            pne = rupture.get_probability_no_exceedance(poes_given_rup_eps)
+            yield (rlzi, poe, imt_str), (iml, pne)
+    else:  # there is a single IML provided by the user; compute the poe
+        poe = numpy.interp(iml, imls, curve_poes)
```

Comment:
You guessed correctly. I need to compute the PoE associated to the IML otherwise the exporter would break. It is a lot easier to compute the PoE here than changing the exporter.